core/src/main/java/org/stellar/anchor/sep10/ISep10Service.java

```diff
@@ -0,0 +1,81 @@
+
+import static org.stellar.sdk.Sep10Challenge.*;
+
+import org.stellar.anchor.api.exception.SepException;
+import org.stellar.anchor.api.sep.sep10.ChallengeRequest;
+import org.stellar.anchor.api.sep.sep10.ChallengeResponse;
+import org.stellar.anchor.api.sep.sep10.ValidationRequest;
+import org.stellar.sdk.Memo;
+
+public interface ISep10Service {
+  /**
+   * Validate the challenge transaction. The default implementation is NOOP. This is used for the
+   * customization of the validation behavior.
+   *
+   * @param request The challenge request.
+   * @throws SepException If the challenge transaction is invalid.
+   */
+  void postChallengeRequestValidation(ChallengeRequest request) throws SepException;
+
+  /**
+   * Validate the memo of the challenge request and return the memo that is validated.
+   *
+   * @param request The challenge request.
+   * @return The validated memo.
+   * @throws SepException If the memo is invalid.
+   */
+  Memo validateChallengeRequestMemo(ChallengeRequest request) throws SepException;
+
+  /**
+   * Validate the client domain of the challenge request.
+   *
+   * @param request The challenge request.
+   * @throws SepException If the client domain is invalid.
+   */
+  void validateChallengeRequestClient(ChallengeRequest request) throws SepException;
+
+  /**
+   * Validate the home domain of the challenge request.
+   *
+   * @param request The challenge request.
+   * @throws SepException If the home domain is invalid.
+   */
+  void validateChallengeRequestFormat(ChallengeRequest request) throws SepException;
+
+  /**
+   * Validate the challenge request. The default implementation is NOOP. This is used for the
+   * customization of the validation behavior.
+   *
+   * @param request The challenge request.
+   */
+  void preChallengeRequestValidation(ChallengeRequest request) throws SepException;
+
+  /** Increment the metrics counter for the number of challenge requests created. */
+  void incrementChallengeRequestCreatedCounter();
+
+  /**
+   * Create a challenge transaction from the request and the memo
+   *
+   * @param request The challenge request.
+   * @param memo The memo.
+   * @return The challenge transaction.
+   * @throws SepException If the challenge transaction cannot be created.
+   */
+  ChallengeResponse createChallengeResponse(ChallengeRequest request, Memo memo)
+      throws SepException;
+
+  /**
+   * Validate the validation request. The default implementation is NOOP. This is used for the
+   * customization of the validation behavior.
+   *
+   * @param request The validation request.
+   * @param challenge The parsed challenge transaction.
+   * @throws SepException If the validation request is invalid.
+   */
+  void preValidateRequestValidation(ValidationRequest request, ChallengeTransaction challenge)
+      throws SepException;
+
+  /** Increment the metrics counter for the number of validation requests validated. */
+  void incrementValidationRequestValidatedCounter();
```

Comment:
This would make Sep10Service abstract and we will need to have another class in the `platform` to implement this function. I am trying to find the balance here.

The `Metrics` class is done by micrometer ana they did a good job in accommodating different metrics frameworks.